pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go

```diff
@@ -1135,7 +1135,9 @@ func (lbaas *LbaasV2) EnsureLoadBalancerDeleted(clusterName string, service *v1.
 		}
 
 		for _, listener := range listenerList {
-			listenerIDs = append(listenerIDs, listener.ID)
+			if len(listener.Loadbalancers) > 0 && listener.Loadbalancers[0].ID == loadbalancer.ID {
```

Comment:
This whole surrounding code should be replaced with a call to `getListenersByLoadBalancerID`